src/test/java/hw6/MyJDITest.java

```diff
@@ -0,0 +1,93 @@
+package hw6;
+
+import com.epam.jdi.light.driver.WebDriverUtils;
+import com.epam.jdi.light.elements.init.PageFactory;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.json.simple.parser.JSONParser;
+import org.json.simple.parser.ParseException;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeSuite;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import org.testng.asserts.SoftAssert;
+
+import java.io.FileReader;
+import java.io.IOException;
+
+import static hw6.JDITestSite.metalsAndColorsPage;
+import static hw6.TestData.DEFAULT_USER;
+import static hw6.JDITestSite.indexPage;
+import static hw6.pages.IndexPage.*;
+import static hw6.pages.MetalsAndColorsPage.*;
+
+public class MyJDITest {
+
+    @BeforeSuite
+    public void beforeSuite() {
+        PageFactory.initSite(JDITestSite.class);
+    }
+
+    @AfterSuite
+    public void afterSuite() {
+        WebDriverUtils.killAllSeleniumDrivers();
+    }
+
+    @Test(dataProvider = "dp")
+    public void metalsAndColorsTest(JSONObject metalColor) {
```

Comment:
You mixed lots of different layers actions in one place.
1. All actions with data should be performed an the data layer, not in the test. You need to pass here as an arguments only the Entity (MetalsAndColors) rather then JSONObject.
2. All actions with Page Elements should be performed in PO layer or in Actions layer, it's up to you.

In the other words, your test sould looks like this:
Site.login(user)
Site.openPage(Pages.METALS_AND_COLORS)
Site.metalsAndColorsPage.metalsAndColorsForm.submit(metalsAndColors)
Site.metalsAndColorsPage.checkLog(metalsAndColors)

Reply:
fixed